src/error/metadata/code.rs

```diff
@@ -1,11 +1,147 @@
+            "E017" => Ok(Code::E017),
+            "E018" => Ok(Code::E018),
+            "E019" => Ok(Code::E019),
+            "E020" => Ok(Code::E020),
+            "E021" => Ok(Code::E021),
+            "E022" => Ok(Code::E022),
+            "E023" => Ok(Code::E023),
+            "E024" => Ok(Code::E024),
+            "E025" => Ok(Code::E025),
+            "E026" => Ok(Code::E026),
+            "E027" => Ok(Code::E027),
+            "E028" => Ok(Code::E028),
+            _ => Err(anyhow!("Invalid error code. Error codes are in the format `E###`"))
+        }
+    }
+}
+
+impl Code {
+    // builds a BTreeMap of every possible code and its explanation, so we can
+    // access from the `explain` function and get a single one OR so we can
+    // iterate over them in `explain_all` for creating a docs page
+    fn explanations() -> BTreeMap<Code, String>{
+        let contents = vec![
+            (Code::E001, include_str!("./codes/E001.md").to_string()),
+            (Code::E002, include_str!("./codes/E002.md").to_string()),
+            (Code::E003, include_str!("./codes/E003.md").to_string()),
+            (Code::E004, include_str!("./codes/E004.md").to_string()),
+            (Code::E005, include_str!("./codes/E005.md").to_string()),
+            (Code::E006, include_str!("./codes/E006.md").to_string()),
+            (Code::E007, include_str!("./codes/E007.md").to_string()),
+            (Code::E008, include_str!("./codes/E008.md").to_string()),
+            (Code::E009, include_str!("./codes/E009.md").to_string()),
+            (Code::E010, include_str!("./codes/E010.md").to_string()),
+            (Code::E011, include_str!("./codes/E011.md").to_string()),
+            (Code::E012, include_str!("./codes/E012.md").to_string()),
+            (Code::E013, include_str!("./codes/E013.md").to_string()),
+            (Code::E014, include_str!("./codes/E014.md").to_string()),
+            (Code::E015, include_str!("./codes/E015.md").to_string()),
+            (Code::E016, include_str!("./codes/E016.md").to_string()),
+            (Code::E017, include_str!("./codes/E017.md").to_string()),
+            (Code::E018, include_str!("./codes/E018.md").to_string()),
+            (Code::E019, include_str!("./codes/E019.md").to_string()),
+            (Code::E020, include_str!("./codes/E020.md").to_string()),
+            (Code::E021, include_str!("./codes/E021.md").to_string()),
+            (Code::E022, include_str!("./codes/E022.md").to_string()),
+            (Code::E023, include_str!("./codes/E023.md").to_string()),
+            (Code::E024, include_str!("./codes/E024.md").to_string()),
+            (Code::E025, include_str!("./codes/E025.md").to_string()),
+            (Code::E026, include_str!("./codes/E026.md").to_string()),
+            (Code::E027, include_str!("./codes/E027.md").to_string()),
+            (Code::E028, include_str!("./codes/E028.md").to_string()),
+        ];
+        BTreeMap::from_iter(contents.into_iter())
```

Comment:
nit: this could likely be a `HashMap` since we don't care about ordering at all

Reply:
I did this so iterating over the codes and printing them in numerical order would work. I'lll take a look and see if Strum could make this easier :)